### PR TITLE
Nodejs compatibility

### LINKED ISSUE #17: Build great new version of cloudproof demo ⭐
Using all the Js elements that we have now (CoverCrypt & Findex) - we can build a great new demo - following these ideas : 
https://docs.google.com/presentation/d/1ZWqUdz59NOVwTjiWdtDAK-5vL_VbkvX24SL6oeFDivI/edit#slide=id.g11fc71b860d_2_29

### LINKED ISSUE #23: Delete wasm_lib directory
Currently, for Jest test purpose, WASM libraries - build for nodejs target - are commited in `tests/wasm_lib`. Find a way to install those dependencies as devdependencies (without breaking Jest tests)

### LINKED ISSUE #29: Run a post-deploy test on demo-cloudproof.cosmian.com
Make sure the demo is working properly

### PR DESCRIPTION
**Pull request by ThibaudDauce**
### Added

- NodeJS support
- encrypt/decrypt with the KMS

### Changed

- WASM files are now base64 inline in the lib
- `KmipClient` is now `KmsClient`
- Removed `client.post()` function, please use the specific functions on the `KmsClient` object.
- Switch from Jest to Vitest
- Refactor the KMS structures (snake case to camel case for properties, change some types…)

### Fixed

- `AccessPolicy.toKmipJson` and `AccessPolicy.toVendorAttribute` are now async (because they call the WASM)

### Fixed

Closes #17 
Closes #23 
Closes #29

### CODE REVIEW

File: src/kms/kmip.ts [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+  OpaqueDataType,
+  SecretDataType,
+  SplitKeyMethod,
+
+  KeyWrapType,
+  RevocationReasonEnumeration,
+}
+
+// Some enums are enums flags. Their implementation differ because a value doesn't have a mandatory corresponding string.
+// For exemple 12 correspond to CryptographicUsageMask.Encrypt | CryptographicUsageMask.Decrypt (4 + 8).
```
Comment (Manuthor) [inline src/kms/kmip.ts:149]:
```suggestion
// For example 12 correspond to CryptographicUsageMask.Encrypt | CryptographicUsageMask.Decrypt (4 + 8).
```

File: src/kms/kmip.ts [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+  // Uint8Array are represented hex encoded.
+  if (kmip instanceof Uint8Array) {
+    if (tag === null)
+      throw new Error(
+        `Trying to TTLV '${kmip.toString()}' but no tag provided.`,
+      )
+    return { tag, type: TtlvType.ByteString, value: hexEncode(kmip) }
+  }
+
+  // Arrays are Structure in KMIP. We recursivly transform each element.
```
Comment (Manuthor) [inline src/kms/kmip.ts:467]:
```suggestion
  // Arrays are Structure in KMIP. We recursively transform each element.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,22 @@
 
 All notable changes to this project will be documented in this file.
 
+## Unrealeased
```
Comment (Manuthor) [inline CHANGELOG.md:5]:
```suggestion
## Unreleased
```

File: rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        env === "node"
+          ? { maxFileSize: 0, targetEnv: "node" }
+          : { targetEnv: "auto-inline" }
+      ),
+    typescript({ outDir: outdir(fmt, env), rootDir: "src", sourceMap: !production, }),
+    {
+      name: "copy-pkg",
+
+      // wasm-bindgen outputs a import.meta.url when using the web target.
+      // rollup will either perserve the the statement when outputting an esm,
```
Comment (Manuthor) [inline rollup.config.js:38]:
```suggestion
      // rollup will either preserve the the statement when outputting an esm,
```

File: src/kms/kmip.ts [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+ *
+ * @param {Serializable} kmip JS KMIP struct
+ * @param {string} tag tag to use
+ * @returns {TTLV} TTLV object
+ */
+export function toTTLV(
+  kmip: Serializable,
+  tag: string | null = null,
+): TTLV {
+  // String are serilize to TextString or Enumeration
```
Comment (Manuthor) [inline src/kms/kmip.ts:389]:
```suggestion
  // String are serialize to TextString or Enumeration
```

**Review by Manuthor — APPROVED** [review]
Great simplifications! no more Jest!

File: package.json [hunk 6]
```diff
@@ -1,55 +1,66 @@
...
-    "ts-jest": "^28.0.8",
-    "ts-loader": "^9.3.1",
-    "typescript": "^4.9.0",
-    "uuid": "^9.0.0",
-    "webpack": "^5.74.0",
-    "webpack-cli": "^4.10.0",
-    "webpack-dev-server": "^4.11.1"
-  }
+    "name": "cloudproof_js",
+    "version": "5.1.0",
```
[the remark this answers is not in the document]
Comment (ThibaudDauce) [reply package.json:3]: Will do this when we release, no?

**Pull request merged**